content/en/docs/instrumentation/java/resources.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Resources
+weight: 70
+cSpell:ignore: getenv myhost SIGINT uuidgen WORKDIR
+---
+
+A [resource](/docs/specs/otel/resource/sdk/) represents the entity producing
+telemetry as resource attributes. For example, a process producing telemetry
+that is running in a container on Kubernetes has a Pod name, a namespace, and
+possibly a deployment name. All three of these attributes can be included in the
+resource.
+
+In your observability backend, you can use resource information to better
+investigate interesting behavior. For example, if your trace or metrics data
+indicate latency in your system, you can narrow it down to a specific container,
+pod, or Kubernetes deployment.
+
+If you use the Javaagent for
+[automatic instrumentation](/docs/instrumentation/java/automatic) you can learn
+how to setup resource detection following the [Agent Configuration
+Guide]/docs/instrumentation/java/automatic/agent-config)
+
+For manual instrumentation, you will find some introductions on how to set up
+resource detection below.
```

Comment:
There's a whole suite of resource providers that detect / add resource attributes based on environment: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/resources/library/README.md

These can be used with or without autoconfigure, and are built into the otel java agent.

Reply:
Add a dependency on the artifact, and use the public APIs to merge various environment based resources with your resource. E.g. to include the [OsResource](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/849e0d5229a1085a80de3867b7a013cf26aaa8d6/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/OsResource.java#L24):
```
Resource resource = Resource.getDefault()
  .merge(OsResource.get())
  .merge(Resource.create(Attributes.builder()
    .put(SERVICE_NAME, "dice-service")
    // ..
    .builder()));
```

Reply:
Ok, that's something we have not described at all ([see my other comment as well](https://github.com/open-telemetry/opentelemetry.io/pull/3144#discussion_r1317066398)). If I understandit correctly you can use those resource providers only if you use the agent? The agent already comes with the SDK and everything included, so I guess the dependency file will look very different then?

Reply:
> If I understandit correctly you can use those resource providers only if you use the agent? The agent already comes with the SDK and everything included, so I guess the dependency file will look very different then?

No, it's a "normal" library instrumentation module: https://mvnrepository.com/artifact/io.opentelemetry.instrumentation/opentelemetry-resources/1.29.0-alpha
It is included by default with the agent, but you can also use it if you configure the SDK manually or via autoconfigure.

Reply:
Ah, okay, then maybe I misunderstood that, let me double check on that :)

Reply:
I added a section, see below